### ISSUE TITLE
The SHOT implementation is a copy of Tent -- possibly a copy paste error. Can you please fix?

### ISSUE THREAD

**Issue by abhirupju** [Yushu-Li/HILTTA#1]

**Comment by Yushu-Li** [comment]
Hi @abhirupju, thanks for pointing this out! I’ve double-checked the implementation — SHOT and TENT are actually different.
•	TENT minimizes prediction entropy directly.
•	SHOT, on the other hand, combines
	1.an entropy term for confidence regularization,
	2.a diversity term encouraging balanced predictions, and
	3.a pseudo-label cross-entropy loss, applied only when the predicted probability exceeds a confidence threshold (β).
Here’s the key part of the SHOT loss:

```
def loss(self, outputs):
    # (1) entropy
    ent_loss = softmax_entropy(outputs).mean(0)

    # (2) diversity
    softmax_out = F.softmax(outputs, dim=-1)
    msoftmax = softmax_out.mean(dim=0)
    ent_loss += torch.sum(msoftmax * torch.log(msoftmax + 1e-5))

    # (3) pseudo label
    py, y_prime = F.softmax(outputs, dim=-1).max(1)
    flag = py > self.beta
    clf_loss = F.cross_entropy(outputs[flag], y_prime[flag])

    loss = ent_loss + self.theta * clf_loss
    return loss
```
Please let me know if you spotted a specific overlap I might have missed.
